crates/aptos-crypto/src/webauthn/webauthn_p256_sigs.rs

```diff
@@ -0,0 +1,238 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+//! This file implements traits for WebAuthn signatures over NIST-P256.
+
+use crate::webauthn::webauthn_traits::WebAuthnSignature;
+use crate::{
+    hash::CryptoHash,
+    p256::P256Signature,
+    traits::*,
+    webauthn::{webauthn_p256_keys::WebAuthnP256PrivateKey, WebAuthnP256PublicKey},
+};
+use anyhow::anyhow;
+use aptos_crypto_derive::{DeserializeKey, SerializeKey};
+use core::convert::TryFrom;
+use serde::Serialize;
+use signature::Verifier;
+use std::fmt;
+use webauthn_rs_core::assertion::parse_public_key_credential;
+use webauthn_rs_core::proto::{COSEAlgorithm, PublicKeyCredential};
+
+/// A WebAuthn P256 signature >= 165 bytes, <= 200 bytes
+#[derive(DeserializeKey, Clone, SerializeKey)]
+pub struct WebAuthnP256Signature {
+    /// Challenge from [`PublicKeyCredential`](webauthn_rs_core::proto::PublicKeyCredential)
+    /// Should be equal to the SHA3-256 digest of the `RawTransaction` -> 32 bytes
+    pub(crate) actual_challenge: Vec<u8>,
+    /// Binary concatenation of:
+    /// 1. [`authenticator_data_bytes`](webauthn_rs_core::internals::AuthenticatorAssertionResponse) and
+    /// 2. SHA-256 digest of [`client_data_bytes`](webauthn_rs_core::internals::AuthenticatorAssertionResponse)
+    ///
+    /// Note: This is compatible with signatures generated by FIDO U2F
+    /// authenticators. See §6.1.2 FIDO U2F Signature Format Compatibility
+    /// https://www.w3.org/TR/webauthn-2/#sctn-fido-u2f-sig-format-compat
+    ///
+    /// authenticator_data_bytes consists of:
+    /// - rpIdHash -> 32 bytes
+    /// - flags -> 1 byte
+    /// - signCount -> 4 bytes
+    /// - attestedCredentialData -> 0 bytes (not included in authenticatorGetAssertion §6.3.3)
+    /// - extensions -> variable bytes (see [`RequestAuthenticationExtensions`](webauthn_rs_core::proto::RequestAuthenticationExtensions))
+    ///
+    /// Total size -> 37 + 32 >= 69 bytes
```

Comment:
Nah greater than or equal to is correct but I dont know how to represent the fact that the size of `authenticator_data_bytes` is `>= 37`. This is because of line 41: `extensions -> variable bytes`

Reply:
I just meant the way it's written sounds like you're literally saying that 37 plus 32 is greater than or equal to 69, which is a trivially correct statement about 3 positive integers

Reply:
Ok i can just say `Total size >= 69 bytes` then

Reply:
That works. I think my original suggestion is more descriptive though

Reply:
🤪 ok sounds good